### PR TITLE
Loop task

### PR DESCRIPTION
**Pull request by MasterOfHash**
Add ability to create looping tasks. These tasks can be used to cause the previous N steps to repeat Y times, or indefinitely (until failure) if the loop value = -1

### CODE REVIEW

File: AIO/Techdebt.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Tech Debt
+- exchangeContext is deprecated
```
Comment (Explv) [inline AIO/Techdebt.md]: Can we remove this file?

Although `exchangeContext` is "deprecated" in the OSBot API, it isn't truly deprecated, it's marked for internal use, but there is no reason that script developers cannot user it.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.PHONY: migrate
+
+migrate:
+	cp out/artifacts/aio/aio.jar ~/OSBot/Scripts/
```
Comment (Explv) [inline Makefile]: Can you remove this file?

File: AIO/src/org/aio/script/AIO.java [hunk 3]
```diff
@@ -133,6 +218,22 @@ private void executeTasks() throws InterruptedException {
         }
     }
 
+    /**
+     * Given the start and end indices, pull a new task queue for the current loop
+     *
+     * @return a queue of tasks
+     */
+    private Queue<Task> getLoopTaskQueue(){
+        ArrayList<Task> allOrderedTasks = gui.getTasksAsList();
```
Comment (Explv) [inline AIO/src/org/aio/script/AIO.java]: Would it be better to just call this once in `loadTasksFromGui`, and store it as a global variable? I don't think we need to do this every time

**Comment by Explv** [comment]
You've done a great job here, I'm very impressed at how you have managed to pick up the code base so quickly and make these relatively complex changes. 👍

I think we can separate out the loop task logic from the main `Script` class, but I will do this in a future commit.

File: AIO/Techdebt.md (hunk 1 above)
Comment (MasterOfHash) [reply AIO/Techdebt.md]: Oh okay good. I asked in chat but no one had a good answer. Do you ever use the chat?

File: Makefile (hunk 2 above)
Comment (MasterOfHash) [reply Makefile]: Sure I'll add it to the .gitignore, I use it to speed up local development.

File: AIO/src/org/aio/script/AIO.java (hunk 3 above)
Comment (MasterOfHash) [reply AIO/src/org/aio/script/AIO.java]: Actually we need a whole new instance of the task each time we execute the loop, that way the task and activity "state" don't leak over into the future executions.

Comment (Explv) [reply AIO/src/org/aio/script/AIO.java]: Good point.

File: AIO/Techdebt.md (hunk 1 above)
Comment (Explv) [reply AIO/Techdebt.md]: Not so much recently

**Pull request merged**